### ISSUE TITLE
tox4 support

### ISSUE THREAD

**Issue by deepyaman** [tox-dev/tox-conda#156]
Will `tox-conda` support `tox4`? I see that there's a `tox-4` branch, but it's almost 100 commits behind `master`/doesn't look maintained.

**Comment by AntoineD** [comment]
`tox4` will be supported, but for now I cannot give a better estimation than during 2023...

**Comment by deepyaman** [comment]
@AntoineD Thanks for the very quick reply! Maybe we'll stick to `tox3` for the time being then, and keep an eye on this.

**Comment by tibortakacs** [comment]
@AntoineD As part of my project's custom tox plugin, I have finished a tox4 plugin with conda support. I am happy to adjust it to make general and be part of this repository. I am not saying that it will immediately cover all features `tox-conda` already has, but I believe it covers the main use cases and could be the foundation of further works for tox4 support. It uses `conda run` instead of activation, it creates a conda `ToxEnv`, so using only one plugin (register new tox env) hook was enough.
Would you consider this as a contribution in `tox-conda`?

**Comment by tibortakacs** [comment]
@AntoineD Of course, happy to. Due to other commitments, it will take some time, I try to make it in two weeks.

**Comment by tibortakacs** [comment]
@AntoineD I have been working on this right now. You can track the progress here: https://github.com/tibortakacs/tox-conda/tree/tox4
I hope I can create a draft PR this week that shows the basic design principles. This can be used as a base of further discussion and work.

**Comment by tibortakacs** [comment]
Please find the draft PR here: https://github.com/tox-dev/tox-conda/pull/163
(This is based on this branch: https://github.com/tibortakacs/tox-conda/tree/tox4)

The main purpose is to agree on the direction, and once there is a common agreement, this PR can be improved by updating the tests and hitting the module hard to find and fix bugs. I have tested with the basic functionalities, it works fine, but I am sure that there are use cases that this PR does not cover yet.

Looking forward to your feedback.

**Comment by tibortakacs** [comment]
@gaborbernat Do you have any suggestion how we can move forward? It might be a good step to use the `tox4` branch, merge this in, and gradually do the improvements and testing. We can also create pre-release package using it.

**Comment by gaborbernat** [comment]
Sure, I think ultimately blocked this on the maintainer (not me) to take it up.

**Comment by tibortakacs** [comment]
Thank you, @gaborbernat. (The PR is automatically assigned to you, I thought you are the one to make decision in this regard.) I will wait for the maintainers' feedback.

**Comment by tibortakacs** [comment]
> Is there an update on tox-conda supporting tox4?

Yes, please find the PR here: https://github.com/tox-dev/tox-conda/pull/163

You can use the merging branch for now, it works fully. I hope I can find time soon to finish off the PR, and merge it back to main.

**Comment by Czaki** [comment]
Package is archived on pypi. 
Is it an official decision? 

@drdavella ?

![Image](https://github.com/user-attachments/assets/fb2cfb20-0c1c-4984-ac0b-bc3f46e7dcd3)

**Comment by drdavella** [comment]
@Czaki I have not been involved in the development of this plugin for quite some time. I did not archive the package but trust that the tox community had a good a reason for doing so.

**Comment by Czaki** [comment]
Did you know who is the proper person to ping? You are the only one person mentioned by name on pypi.
